deid-hbase-phoenix/src/main/scala/SparkPhoenixDeid.scala

```diff
@@ -0,0 +1,109 @@
+  * NOTE: for Phoenix, the hbase-site.xml file MUST be copied into spark/conf
+  * cp /usr/hdp/current/hbase-client/conf/hbase-site.xml /usr/hdp/current/spark2-client/conf/
+  */
+
+// singleton to provide single Phoenix connection per executor
+object PhoenixConnectionSingleton {
+  @transient private var connection: Connection = _
+
+  def getConnectionInstance(jdbcUrl: String): Connection = {
+    if (connection == null) {
+      // need a connection to Phoenix for each executor
+      connection = DriverManager.getConnection(jdbcUrl)
+      print("connection created")
+    }
+    connection
+  }
+
+}
+
+object SparkPhoenixDeid {
+  def main(args: Array[String]) = {
+
+    if (args.length < 1) {
+      System.err.println(
+        s"""
+           |Usage: spark-submit ... <Schema.Sequence>
+           |  <Schema.Sequence> Phoenix schema.sequence to read
+           |
+        """.stripMargin)
+      System.exit(1)
+    }
+
+    val Array(sequenceSchema) = args
+
+
+    val sparkConf = new SparkConf()
+    // hbase master should be read from hbase-site.xml
+    //sparkConf.set("hbase.master", "hdp-ambari.internal.decisionlab.io"+":16000")
+
+    val spark = SparkSession.builder()
+      .config(sparkConf)
+      .getOrCreate()
+
+    // This list must only contain de-duplicated Patient IDs that need a new DeID
+    val patientInfoRdd = spark.sparkContext.parallelize(Array(
+      ("1", "p0001"),
+      ("2", "p0002"),
+      ("3", "p0003"),
+      ("4", "p0004"),
+      ("5", "p0005"),
+      ("6", "p0006"),
+      ("7", "p0007"),
+      ("8", "p0008")
+    ))
+
+    import spark.implicits._
+    var patientInfoDf = patientInfoRdd.toDF("id", "pid")
+
+    // function that queries HBase for DeID matching patient ID
+    def queryDeid = (id: String) => {
+      var deid: Integer = null
+
+      // hbase master and znode.parent will be read from hbase-site.xml
+      val jdbcUrl = "jdbc:phoenix" //:hdp-ambari.internal.decisionlab.io:2181:/hbase-unsecure"
+      val sequenceQuery = "SELECT NEXT VALUE FOR " + sequenceSchema //ahinchliff_schema.seq_test"
+      val connection = PhoenixConnectionSingleton.getConnectionInstance(jdbcUrl)
+
+      // don't see how to use a 'SELECT' statement with spark.read.jdbc() ?
+      //val connectionProperties = new Properties()
+      //val sequenceDF = spark.read.jdbc(jdbcUrl, sequenceQuery, connectionProperties)
+
+      val pstmt = connection.prepareStatement(sequenceQuery)
+      val rs: ResultSet = pstmt.executeQuery
+
+      rs.next()
+      deid = rs.getInt(1)
+
+      if (rs.next()){
+        throw new SQLException("Programming Error: ResultSet should have only returned one value!")
+      }
+
+      deid
+    }
+    val deidUdf = udf(queryDeid)
+
+    // add new column to patient event data with DeID
+    patientInfoDf = patientInfoDf.withColumn("deid", deidUdf(col("pid")))
+
+    // write out for review
+    patientInfoDf.rdd.saveAsTextFile("sampleDeid")
```

Comment:
I noticed that I ended up with two part files, both with 4 entires each. If you wanted to, you could repartition(1) here and end up with only 1 file